src/components/content/order/CreateService.tsx

```diff
@@ -107,44 +110,87 @@ function CreateService(): JSX.Element {
         }
         setCategoryName(categoryName);
         setServiceName(serviceName);
-        void serviceVendorApi.listRegisteredServices(categoryName, '', serviceName, '').then((rsp) => {
-            if (rsp.length > 0) {
-                const currentVersions: Map<string, RegisteredServiceVo[]> = new Map<string, RegisteredServiceVo[]>();
-                for (const registerServiceEntity of rsp) {
-                    if (registerServiceEntity.version) {
-                        if (!currentVersions.has(registerServiceEntity.version)) {
-                            currentVersions.set(
-                                registerServiceEntity.version,
-                                rsp.filter((data) => data.version === registerServiceEntity.version)
-                            );
+        void serviceVendorApi
+            .listRegisteredServices(categoryName, '', serviceName, '')
+            .then((rsp) => {
+                if (rsp.length > 0) {
+                    const currentVersions: Map<string, RegisteredServiceVo[]> = new Map<
+                        string,
+                        RegisteredServiceVo[]
+                    >();
+                    for (const registerServiceEntity of rsp) {
+                        if (registerServiceEntity.version) {
+                            if (!currentVersions.has(registerServiceEntity.version)) {
+                                currentVersions.set(
+                                    registerServiceEntity.version,
+                                    rsp.filter((data) => data.version === registerServiceEntity.version)
+                                );
+                            }
                         }
                     }
+                    versionMapper.current = currentVersions;
+                    initialServiceValue(latestVersion);
+                } else {
+                    return;
                 }
-                versionMapper.current = currentVersions;
+            })
+            .catch((error: Error) => {
+                console.log(error.message);
+                versionMapper.current = new Map<string, RegisteredServiceVo[]>();
+            });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [location, urlParams]);
```

Comment:
Do we need `location` as a dependence here? Maybe urlParams is enough. 
But `initialServiceValue` is not suitable to be a dependence, can we do something to avoid using a function?

Reply:
fixed.